### PR TITLE
set INTERCEPT_ALL_OBJS in tests when needed

### PR DESCRIPTION
**Pull request by GBuella**
This allows testing build with the -fsanitize=address flag.
The INTERCEPT_ALL_OBJS is not needed during asm_pattern tests,
is needed during logging tests (as some log messages might pass
through syscall instructions in asan object code outside libc),
and are not needed during the prog_* tests - as those tests
only look at a specific syscall made in the app anyways.

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/syscall_intercept/28)

### CODE REVIEW

**Comment by krzycz** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/syscall_intercept/28#-:-KnnTqRjPXzG0pxj3Czj:bnfp4nl)*

**Comment by GBuella** [comment]
With this patch, and the one in https://github.com/pmem/nvml/pull/2063 (which I'll bring downstream from NVML after being merged there) I managed to successfully execute all tests with clang and ASAN.

**Comment by GBuella** [comment]
Update: I added another commit, for setting another variable.

**Comment by krzycz** [comment]
Reviewed 4 of 4 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/syscall_intercept/28)*

File: test/check_log.cmake [hunk 1]
```diff
@@ -45,18 +45,28 @@ endif()
...
+	if(TEST_EXTRA_PRELOAD)
+		set(ENV{LD_PRELOAD} ${TEST_EXTRA_PRELOAD}:${LIB_FILE})
+	else()
+		set(ENV{LD_PRELOAD} ${LIB_FILE})
+	endif()
 	execute_process(COMMAND ${TEST_PROG}
 		${TEST_PROG_ARG} ${LOG_OUTPUT} ${SECOND_LOG_OUTPUT}
 		RESULT_VARIABLE HAD_ERROR)
 	unset(ENV{LD_PRELOAD})
+	unset(ENV{LD_DEBUG})
```
Comment (marcinslusarz) [inline test/check_log.cmake]: Is this supposed to be here?

File: test/check_log.cmake [hunk 2]
```diff
@@ -45,18 +45,28 @@ endif()
...
-	set(ENV{LD_PRELOAD} ${LIB_FILE})
+	if(TEST_EXTRA_PRELOAD)
+		set(ENV{LD_PRELOAD} ${TEST_EXTRA_PRELOAD}:${LIB_FILE})
+	else()
+		set(ENV{LD_PRELOAD} ${LIB_FILE})
+	endif()
 	execute_process(COMMAND ${TEST_PROG} ${TEST_PROG_ARG} ${LOG_OUTPUT}
 		RESULT_VARIABLE HAD_ERROR)
 	unset(ENV{LD_PRELOAD})
+	unset(ENV{LD_DEBUG})
```
Comment (marcinslusarz) [inline test/check_log.cmake]: ditto

File: test/check_log.cmake (hunk 1 above)
Comment (GBuella) [inline test/check_log.cmake]: Nope, I just forgot it.

File: test/check_log.cmake (hunk 2 above)
Comment (GBuella) [inline test/check_log.cmake]: Removed.

File: test/check_log.cmake (hunk 1 above)
Comment (GBuella) [inline test/check_log.cmake]: Removed.

**Comment by marcinslusarz** [comment]
:lgtm:

---

Reviewed 3 of 4 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/syscall_intercept/28#-:-KnxLdAUVUAJeAhVFVvg:bnfp4nl)*

**Comment by sarahjelinek** [comment]
:lgtm: I am good with this.  Just a question. Is there a way to do an 'include' in a CMakeLists.txt file? It seems silly to have to continue to repeat the set the same variable over and over.

---

Reviewed 3 of 4 files at r3, 1 of 1 files at r4.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/syscall_intercept/28#-:-Ko7xObYJ5qkIn1T2qkn:b-9fxrh)*

**Pull request merged**